### PR TITLE
[refactor] Creates role service for querying available roles

### PR DESCRIPTION
**Pull request by marcoviappiani**
This creates a roles service and refactors the tab-related controller to call this service for querying available roles.
#201

### CODE REVIEW

**Comment by marcoviappiani** [comment]
@RRDuke looking forward to your feedback!

File: modules/core/client/controllers/editTabDialogue.controller.client.js [hunk 1]
```diff
@@ -17,9 +17,8 @@ function EditTabDialogueController ($mdDialog, allUsersService, tab, tabsService
   vm.title = '';
 
   // Add role autocomplete variables below
-  vm.availableRoles = [];
-  vm.querySearch = querySearch;
   vm.searchText = null;
+  vm.querySearch = rolesService.queryAvailableRoles;
```
Comment (codydaig) [inline modules/core/client/controllers/editTabDialogue.controller.client.js:20]: Alphabetical order

File: modules/core/client/services/roles.service.client.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+angular.module('nobsjs')
+  .factory('rolesService', rolesService);
+
+rolesService.$inject = ['allUsersService', 'tabsService'];
+
+function rolesService (allUsersService, tabsService) {
+  var availableRoles;
+  var roles = {
+    queryAvailableRoles: queryAvailableRoles,
+    getAvailableRoles: getAvailableRoles,
```
Comment (codydaig) [inline modules/core/client/services/roles.service.client.js:11]: Alphabetical Order

**Comment by marcoviappiani** [comment]
added a commit to refactor also the edituser controller

**Pull request merged**